src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h

```diff
@@ -1129,10 +1173,45 @@ namespace UiaOperationAbstraction
                 static_assert(always_false<T>::value, "Unexpected array element comparison type.");
             }
         }
+
+        template <std::size_t I, typename... ItemWrapperType>
+        inline void LocalTupleToRemoteArray(
+            [[maybe_unused]] const std::tuple<typename ItemWrapperType::LocalType...>& localTuple,
+            [[maybe_unused]] winrt::Microsoft::UI::UIAutomation::AutomationRemoteArray& remoteArray)
+        {
+            if constexpr (I < sizeof...(ItemWrapperType))
+            {
+                using TupleItemWrapperType = typename UiaTuple<ItemWrapperType...>::TupleItemWrapperType<I>;
+
+                TupleItemWrapperType wrapper{ std::get<I>(localTuple) };
+                wrapper.ToRemote();
+                remoteArray.Append(wrapper);
+
+                LocalTupleToRemoteArray<I + 1, ItemWrapperType...>(localTuple, remoteArray);
+            }
+        }
+
+        template <std::size_t I, typename... ItemWrapperType>
+        inline void RemoteArrayToLocalTuple(
+            [[maybe_unused]] const winrt::Windows::Foundation::Collections::IVector<winrt::Windows::Foundation::IInspectable>& remoteArray,
+            [[maybe_unused]] std::tuple<typename ItemWrapperType::LocalType...>& localTuple)
+        {
+            if constexpr (I < sizeof...(ItemWrapperType))
+            {
+                using TupleItemWrapperType = typename UiaTuple<ItemWrapperType...>::TupleItemWrapperType<I>;
+                using ItemLocalType = typename TupleItemWrapperType::LocalType;
+
+                TupleItemWrapperType wrappedLocal = ItemLocalType();
+                wrappedLocal.FromRemoteResult(remoteArray.GetAt(I));
+                std::get<I>(localTuple) = static_cast<ItemLocalType>(wrappedLocal);

```

Comment:
>static_cast<ItemLocalType> [](start = 42, length = 26)

Is it better to use `std::get<ItemLocalType>` for `std::variant` type?
I mimic the logic from `UiaArray::FromRemoteResult`, but shall we sue `std::get`?

Reply:
We cannot. This is a wrapper type. `m_member`  is internal to the wrapper.

---
In reply to: [504973773](https://github.com/microsoft/Microsoft-UI-UIAutomation/pull/43#discussion_r504973773) [](ancestors = 504973773)